dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/TaggedEvent.java

```diff
@@ -0,0 +1,39 @@
+package com.dat3m.dartagnan.program.event.core;
+
+import com.dat3m.dartagnan.program.event.Tag;
+import com.dat3m.dartagnan.program.event.visitors.EventVisitor;
+
+public class TaggedEvent extends AbstractEvent {
+
+    protected final String name;
+
+    public TaggedEvent(String name, String ... tags) {
+        this.name = name;
+        this.addTags(Tag.VISIBLE, name);
+        this.addTags(tags);
+    }
+
+    protected TaggedEvent(TaggedEvent other) {
+        super(other);
+        this.name = other.name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String defaultString() {
+        return name;
+    }
```

Comment:
I don't think we should do that because that is mixing two things. Whatever is chosen for a name should be informative enough to understand the event semantics.